File: lib/model/native.dart
class NativeParam {
  final String type;

  final String name;

  NativeParam(this.type, this.name);
}

class Native {
  final String name;

  final String hash;

  final String? jhash;

  final String comment;

  final List<NativeParam> params;

  final String returnType;

  final String build;

  final List<String> oldNames;

  Native(this.name, this.hash, this.jhash, this.comment, this.params,
      this.returnType, this.build, this.oldNames);
}

File: spa/src/store/model/Native.ts
import CodeExample from './CodeExample';
import NativeParam from './NativeParam'

export default interface Native {
  namespace : string
  name      : string
  hash      : string
  comment   : string
  params    : NativeParam[]
  returnType: string
  oldNames ?: string[]
  build    ?: string
  jhash    ?: string
  apiSet   ?: string
  examples ?: CodeExample[]
}
